src/annotator/guest.js

```diff
@@ -241,21 +243,10 @@ export default class Guest {
    * Retrieve metadata for the current document.
    */
   getDocumentInfo() {
-    let metadataPromise;
-    let uriPromise;
-
-    if (this.pdfIntegration) {
-      metadataPromise = this.pdfIntegration.getMetadata();
-      uriPromise = this.pdfIntegration.uri();
-    } else {
-      uriPromise = Promise.resolve(this.documentMeta.uri());
-      metadataPromise = Promise.resolve(this.documentMeta.metadata);
-    }
-
-    uriPromise = uriPromise.catch(() =>
-      decodeURIComponent(window.location.href)
-    );
-    metadataPromise = metadataPromise.catch(() => ({
+    const uriPromise = this.integration
+      .uri()
+      .catch(() => decodeURIComponent(window.location.href));
```

Comment:
Do we want to silently default to `window.location.href` in case of errors? If the correctness of this piece of information is essential for the application it maybe better to not catch the error and make the application fail.

Reply:
I agree. I think changing this is out of scope for this PR but we definitely don't want `window.location.href` being used in PDF documents for example! I filed https://github.com/hypothesis/client/issues/3204 to track this issue.